numba/cuda/codegen.py

```diff
@@ -16,18 +16,7 @@ def _optimize_functions(self, ll_module):
         pass
 
     def _optimize_final_module(self):
-        # Run some lightweight optimization to simplify the module.
-        # This seems to workaround a libnvvm compilation bug (see #1341)
-        pmb = ll.PassManagerBuilder()
-        pmb.opt_level = 1
-        pmb.disable_unit_at_a_time = False
-        pmb.disable_unroll_loops = True
-        pmb.loop_vectorize = False
-        pmb.slp_vectorize = False
-
-        pm = ll.ModulePassManager()
-        pmb.populate(pm)
-        pm.run(self._final_module)
+        pass
```

Comment:
Want to add a note/reference to this PR about why final module optimisation is not performed?

Reply:
I've added a comment that covers both the optimization at the function and module level.